File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/event/search/EventSearchCriteria.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.event.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchOperator;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.event.id.IEventId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.event.search.EventSearchCriteria")
public class EventSearchCriteria extends AbstractObjectSearchCriteria<IEventId>
{

    private static final long serialVersionUID = 1L;

    public EventTypeSearchCriteria withEventType()
    {
        return with(new EventTypeSearchCriteria());
    }

    public EventEntityTypeSearchCriteria withEntityType()
    {
        return with(new EventEntityTypeSearchCriteria());
    }

    public EventEntitySpaceSearchCriteria withEntitySpace()
    {
        return with(new EventEntitySpaceSearchCriteria());
    }

    public EventEntitySpaceIdSearchCriteria withEntitySpaceId()
    {
        return with(new EventEntitySpaceIdSearchCriteria());
    }

    public EventEntityProjectSearchCriteria withEntityProject()
    {
        return with(new EventEntityProjectSearchCriteria());
    }

    public EventEntityProjectIdSearchCriteria withEntityProjectId()
    {
        return with(new EventEntityProjectIdSearchCriteria());
    }

    public EventEntityRegistratorSearchCriteria withEntityRegistrator()
    {
        return with(new EventEntityRegistratorSearchCriteria());
    }

    public EventEntityRegistrationDateSearchCriteria withEntityRegistrationDate()
    {
        return with(new EventEntityRegistrationDateSearchCriteria());
    }

    public EventRegistratorSearchCriteria withRegistrator()
    {
        return with(new EventRegistratorSearchCriteria());
    }

    public EventRegistrationDateSearchCriteria withRegistrationDate()
    {
        return with(new EventRegistrationDateSearchCriteria());
    }

    public EventSearchCriteria withOrOperator()
    {
        return (EventSearchCriteria) withOperator(SearchOperator.OR);
    }

    public EventSearchCriteria withAndOperator()
    {
        return (EventSearchCriteria) withOperator(SearchOperator.AND);
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("EVENT");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/event/search/EventSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/common/search/AbstractObjectSearchCriteria", "as/dto/common/search/SearchOperator", "as/dto/event/search/EventTypeSearchCriteria", "as/dto/event/search/EventEntityTypeSearchCriteria",
 "as/dto/event/search/EventEntitySpaceSearchCriteria", "as/dto/event/search/EventEntitySpaceIdSearchCriteria", "as/dto/event/search/EventEntityProjectSearchCriteria",
 "as/dto/event/search/EventEntityProjectIdSearchCriteria", "as/dto/event/search/EventEntityRegistratorSearchCriteria", "as/dto/event/search/EventEntityRegistrationDateSearchCriteria",
 "as/dto/event/search/EventRegistratorSearchCriteria", "as/dto/event/search/EventRegistrationDateSearchCriteria"],
	function(require, stjs, AbstractObjectSearchCriteria, SearchOperator) {

	var EventSearchCriteria = function() {
		AbstractObjectSearchCriteria.call(this);
	};

	stjs.extend(EventSearchCriteria, AbstractObjectSearchCriteria, [ AbstractObjectSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.event.search.EventSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withEventType = function() {
			var EventEventTypeSearchCriteria = require("as/dto/event/search/EventTypeSearchCriteria");
			return this.addCriteria(new EventEventTypeSearchCriteria());
		};
		prototype.withEntityType = function() {
			var EventEntityTypeSearchCriteria = require("as/dto/event/search/EventEntityTypeSearchCriteria");
			return this.addCriteria(new EventEntityTypeSearchCriteria());
		};
		prototype.withEntitySpace = function() {
			var EventEntitySpaceSearchCriteria = require("as/dto/event/search/EventEntitySpaceSearchCriteria");
			return this.addCriteria(new EventEntitySpaceSearchCriteria());
		};
		prototype.withEntitySpaceId = function() {
			var EventEntitySpaceIdSearchCriteria = require("as/dto/event/search/EventEntitySpaceIdSearchCriteria");
			return this.addCriteria(new EventEntitySpaceIdSearchCriteria());
		};
		prototype.withEntityProject = function() {
			var EventEntityProjectSearchCriteria = require("as/dto/event/search/EventEntityProjectSearchCriteria");
			return this.addCriteria(new EventEntityProjectSearchCriteria());
		};
		prototype.withEntityProjectId = function() {
			var EventEntityProjectIdSearchCriteria = require("as/dto/event/search/EventEntityProjectIdSearchCriteria");
			return this.addCriteria(new EventEntityProjectIdSearchCriteria());
		};
		prototype.withEntityRegistrator = function() {
			var EventEntityRegistratorSearchCriteria = require("as/dto/event/search/EventEntityRegistratorSearchCriteria");
			return this.addCriteria(new EventEntityRegistratorSearchCriteria());
		};
		prototype.withEntityRegistrationDate = function() {
			var EventEntityRegistrationDateSearchCriteria = require("as/dto/event/search/EventEntityRegistrationDateSearchCriteria");
			return this.addCriteria(new EventEntityRegistrationDateSearchCriteria());
		};
		prototype.withRegistrator = function() {
			var EventRegistratorSearchCriteria = require("as/dto/event/search/EventRegistratorSearchCriteria");
			return this.addCriteria(new EventRegistratorSearchCriteria());
		};
		prototype.withRegistrationDate = function() {
			var EventRegistrationDateSearchCriteria = require("as/dto/event/search/EventRegistrationDateSearchCriteria");
			return this.addCriteria(new EventRegistrationDateSearchCriteria());
		};
		prototype.withOrOperator = function() {
			return this.withOperator(SearchOperator.OR);
		};
		prototype.withAndOperator = function() {
			return this.withOperator(SearchOperator.AND);
		};
	}, {
		operator : {
			name : "Enum",
			arguments : [ "SearchOperator" ]
		},
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});

	return EventSearchCriteria;
});
